src/realm/table_view.cpp

```diff
@@ -385,9 +357,17 @@ void TableView::clear()
     bool sync_to_keep =
         m_last_seen_versions == get_dependency_versions() && !m_descriptor_ordering.will_apply_distinct();
 
-    _impl::TableFriend::batch_erase_rows(*get_parent(), m_key_values); // Throws
-
-    m_key_values.clear();
+    // Remove all invalid keys
+    auto it = m_key_values.begin();
+    while (it != m_key_values.end()) {
+        if (m_table->is_valid(*it)) {
+            ++it;
+        }
+        else {
+            it = m_key_values.erase(it);
```

Comment:
remove from the middle, remove_if and erase instead?

Reply:
put it here: https://github.com/realm/realm-core/pull/7030

Reply:
Of course. I have to remember that approach.